src/main/java/seedu/ichifund/logic/commands/repeater/AddRepeaterCommand.java

```diff
@@ -70,18 +75,57 @@ public CommandResult execute(Model model) throws CommandException {
         RepeaterUniqueId repeaterUniqueId = model.getCurrentRepeaterUniqueId();
 
         // Add associated transactions.
-        /*
-        boolean isStartMonth = true;
-        boolean isEndMonth = false;
         int currentMonth = toAdd.getStartDate().getMonth().monthNumber;
         int currentYear = toAdd.getStartDate().getYear().yearNumber;
         int endMonth = toAdd.getEndDate().getMonth().monthNumber;
         int endYear = toAdd.getEndDate().getYear().yearNumber;
 
-        while (currentYear <= endYear && currentMonth <= endMonth) {
-
+        while ((currentYear < endYear) || (currentYear == endYear && currentMonth <= endMonth)) {
+            if (!toAdd.getMonthStartOffset().isEmpty()) {
+                Transaction transaction = new Transaction(
+                        toAdd.getDescription(),
+                        toAdd.getAmount(),
+                        toAdd.getCategory(),
+                        new Date(
+                            new Day(toAdd.getMonthStartOffset().toString()),
+                            new Month(String.valueOf(currentMonth)),
+                            new Year(String.valueOf(currentYear))),
+                        toAdd.getTransactionType(),
+                        repeaterUniqueId);
+                model.addTransaction(transaction);
```

Comment:
This function has side effects, I will need to change the code before you can use this

Reply:
Yeap, definitely